### ISSUE TITLE
Error in postprocessing with coco-constrained data

### ISSUE THREAD

**Issue by brockho** [numbbo/coco#1383]
When postprocessing a freshly generated data set (with scipy's COBYLA), I get the following error:
```
Post-processing (1): will generate output data in folder ppdata0010xD-003\fmin_cobyla_on_bbob-constr
ained_budget0010xD-003
  this might take several minutes.
C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\pproc.py
:963: UserWarning: There is a difference between the maxevals in the *.info file and in the data fil
es.
  warnings.warn('There is a difference between the maxevals in the '
  Data consistent according to test in consistency_check() in pproc.DataSet
Scaling figures...
C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\bestalg.
py:402: UserWarning: no best algorithm data specified
  warnings.warn("no best algorithm data specified")
  done (Tue May 23 22:18:07 2017).
Generating LaTeX tables...
  done (Tue May 23 22:18:09 2017).
ECDF graphs...
Traceback (most recent call last):
  File "C:\Users\dimo\Anaconda2\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Users\dimo\Anaconda2\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\
__main__.py", line 30, in <module>
    rungeneric.main(args)
  File "C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\
rungeneric.py", line 279, in main
    rungeneric1.main(genopts + ["-o", outputdir, alg])
  File "C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\
rungeneric1.py", line 409, in main
    pprldistr.main(sliceNoise, True, algoutputdir, '%s' % noise)
  File "C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\
pprldistr.py", line 879, in main
    **rldUnsuccStyles[j % len(rldUnsuccStyles)])
  File "C:\Users\dimo\AppData\Roaming\Python\Python27\site-packages\cocopp-2.0.258-py2.7.egg\cocopp\
pprldistr.py", line 517, in plotFVDistr
    'was smaller than maximal budget ' + str(budget * ds.dim)
AssertionError: first entry 90.0was smaller than maximal budget 2.5
```

**Comment by brockho** [comment]
The problem was most likely caused by some bugs in the `bbob` logger/observer code in C that only appeared when the constraint function was evaluated before the first time the objective function was called.

Together with @nikohansen, we propose a first fix (in my fork's `dimo-develop` branch) that is currently running the tests on the `devel-test3` branch and should be merged with `development` as soon as the CI tests run through.

**Comment by brockho** [comment]
This bug is now solved with pull request #1386. It remains to update the example experiment in Python to make sense in the constrained case (#1384). However, this issue here can be closed now.
